tests/functional/test_logger.py

```diff
@@ -683,6 +683,29 @@ def handler(event, context):
         assert key in second_log
 
 
+def test_clear_state_keeps_custom_keys(lambda_context, stdout, service_name):
+    # GIVEN
+    date_format = "%Y"
+    location_format = "%(module)s:%(funcName)s:%(lineno)d"
+    logger = Logger(service=service_name, stream=stdout, location=location_format, datefmt=date_format)
+
+    # WHEN clear_state is set
+    @logger.inject_lambda_context(clear_state=True)
+    def handler(event, context):
+        logger.info("Foo")
+
+    # THEN all standard keys should be available as usual
+    handler({}, lambda_context)
+    handler({}, lambda_context)
+
+    first_log, second_log = capture_multiple_logging_statements_output(stdout)
+
+    assert re.fullmatch("[0-9]{4}", first_log["timestamp"])
+    assert re.fullmatch("[0-9]{4}", second_log["timestamp"])
+    assert re.fullmatch(".+:.+:[0-9]+", first_log["location"])
+    assert re.fullmatch(".+:.+:[0-9]+", second_log["location"])
```

Comment:
Can we use an alternative to regex to confirm custom keys are being preserved?

While correct this will slow down tests as we grow them (currently 684), and it might make it difficult for the untrained eye.

For example, we could:

* replace `datefmt=date_format` with any custom key like `foo=bar` and assert on that
* replace `:%(lineno)d` with `clear_state` `location_format` so you can assert using `test_logger.handler:clear_state` - line number doesn't become that important and we continue to assert on that custom location

Thank you!

Reply:
Smart, not a fan of regex so happy to exclude it. Pushing update!